chain/paloma/client.go

```diff
@@ -110,26 +110,73 @@ func (c Client) QueryMessagesForAttesting(ctx context.Context, queueTypeName str
 
 // QueryMessagesForRelaying returns all messages that are currently in the queue.
 func (c Client) QueryMessagesForRelaying(ctx context.Context, queueTypeName string) ([]chain.MessageWithSignatures, error) {
-	return queryMessagesInQueue(
+	return queryMessagesForRelaying(
 		ctx,
 		queueTypeName,
-		nil,
+		c.valAddr,
 		c.GRPCClient,
 		c.L.Codec.Marshaler,
 	)
 }
 
-func queryMessagesInQueue(
+func queryMessagesForRelaying(
 	ctx context.Context,
 	queueTypeName string,
-	skipEvidence sdk.ValAddress,
+	valAddress sdk.ValAddress,
 	c grpc.ClientConn,
 	anyunpacker codectypes.AnyUnpacker,
 ) ([]chain.MessageWithSignatures, error) {
 	qc := consensus.NewQueryClient(c)
-	msgs, err := qc.MessagesInQueue(ctx, &consensus.QueryMessagesInQueueRequest{
-		QueueTypeName:                    queueTypeName,
-		SkipEvidenceProvidedByValAddress: skipEvidence,
+	msgs, err := qc.QueuedMessagesForRelaying(ctx, &consensus.QueryQueuedMessagesForRelayingRequest{
+		QueueTypeName: queueTypeName,
+		ValAddress:    valAddress,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	msgsWithSig := []chain.MessageWithSignatures{}
```

Comment:
Suggestion: since we know the length of what's required in advance, we can actually assign a fixed sized slice already:

`msgsWithSig := make([]chain.MessageWithSignatures, 0, len(msgs.Messages)`

Instead of appending, we can then simply assign via index. Same for the signatures below.

Reply:
Two ways of doing the same thing.  Is one better than the other?  I imagine using an index is more efficient, but I don't know for sure or by how much.

Reply:
Usually, if the size of a data set is not known, I will go the same approach and just use append. If the size is known though, it's become muscle memory for me to assign a fixed size slice and access directly. Not only is it faster, it also avoids potential array resizing (depending on size) and decreases memory footprint. 

To be fair, I think the gains are usually not that noticeable. It does become a factor in high load scenarios though. And ultimately it's basically free optimisation at no cost - so why not do it?

Reply:
Good points.  I'll make the changes :+1: